### PR TITLE
Inflight OAuth request store

### PR DESCRIPTION
**Pull request by peterschwarz**
This PR refactors the in-flight state saved between the initiation of an OAuth2 request and the receipt of the callback. It places the original memory implementation behind a trait, and updates the builders to take a trait implementation. 

Additionally, 

- Adding a new parameter create a clippy error with the OAuthClient constructor. As this is a private method, it has been decomposed into two functions

### CODE REVIEW

File: libsplinter/src/auth/oauth/builder/mod.rs [hunk 1]
```diff
@@ -144,4 +154,14 @@ impl OAuthClientBuilder {
         self.identity_provider = Some(identity_provider);
         self
     }
+
+    /// Sets the in-flight request store to use to store values between requests to and from the
```
Comment (agunde406) [inline libsplinter/src/auth/oauth/builder/mod.rs:154]: I think "to use" should be removed from this comment?

Comment (agunde406) [reply libsplinter/src/auth/oauth/builder/mod.rs:154]: same for other instance of with_inflight_request_store

Comment (peterschwarz) [reply libsplinter/src/auth/oauth/builder/mod.rs]: Reworded in all places.

**Review by agunde406 — APPROVED** [review]

**Review by agunde406 — DISMISSED** [review]

File: libsplinter/src/oauth/mod.rs [hunk 2]
```diff
@@ -49,13 +45,14 @@ const PENDING_AUTHORIZATION_EXPIRATION_SECS: u64 = 3600; // 1 hour
...
     client: BasicClient,
-    /// Pending authorization requests, including the CSRF token, PKCE verifier, and client's
-    /// redirect URL
-    pending_authorizations: Arc<Mutex<TtlMap<String, PendingAuthorization>>>,
     /// The scopes that will be requested for each user that's authenticated
     scopes: Vec<String>,
     /// OAuth2 identity provider used to retrieve users' identities
     identity_provider: Box<dyn IdentityProvider>,
+
+    /// Store for pending authorization requests , including the CSRF token, PKCE verifier, and client's
```
Comment (ltseeley) [inline libsplinter/src/oauth/mod.rs:53]: There's an extra space in this line

File: libsplinter/src/oauth/store/memory.rs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+use std::time::Duration;
+
+use crate::oauth::{InflightOAuthRequestStore, PendingAuthorization};
+use crate::collections::TtlMap;
+use crate::error::InternalError;
+
+/// The amount of time before a pending authorization expires and a new request must be made
+const PENDING_AUTHORIZATION_EXPIRATION_SECS: u64 = 3600; // 1 hour
+
+#[derive(Clone)]
```
Comment (ltseeley) [inline libsplinter/src/oauth/store/memory.rs:27]: Needs a doc string to provide some information about this implementation, especially around the timed expiration

File: libsplinter/src/oauth/store/mod.rs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+// Copyright 2018-2020 Cargill Incorporated
```
Comment (ltseeley) [inline libsplinter/src/oauth/store/mod.rs:1]: We're planning on having more than one store that belongs to OAuth, right? Probably want this module to have a more specific name.

File: libsplinter/src/oauth/mod.rs [hunk 5]
```diff
@@ -74,30 +74,11 @@ impl OAuthClient {
     ///
     /// Returns an error if any of the auth, redirect, or token URLs are invalid
     fn new(
-        client_id: String,
```
Comment (ltseeley) [inline libsplinter/src/oauth/mod.rs:80]: Doc string for this method needs to be updated

File: libsplinter/src/oauth/store/mod.rs (hunk 4 above)
Comment (peterschwarz) [reply libsplinter/src/oauth/store/mod.rs:1]: We can rearrange them when we create the next one

Comment (peterschwarz) [reply libsplinter/src/oauth/store/mod.rs:1]: Or, there should be one OAuth store

**Review by ltseeley — DISMISSED** [review]

**Review by ltseeley — APPROVED** [review]

**Review by agunde406 — APPROVED** [review]

**Pull request merged**